### PR TITLE
Let sphinx add parameter description with nested structures to documentation

### PR DESCRIPTION
**Pull request by christophfroehlich**
This adds automatic documentation generation from generate_parameter_library yamls, where nested structures or `__map_` parameters are used and completes #651 

- [x]  It remains draft until the missing feature of the library will be implemented, see https://github.com/ros-controls/control.ros.org/pull/216

Binary builds will fail until generate_parameter_library is released: the multiline description generates invalid C++ code.

### CODE REVIEW

File: joint_trajectory_controller/doc/parameters_context.yaml [hunk 1]
```diff
@@ -0,0 +1,13 @@
+constraints:
+    Default values for tolerances if no explicit values are states in JointTrajectory message.
```
Comment (fmauch) [inline joint_trajectory_controller/doc/parameters_context.yaml:2]:
```suggestion
    Default values for tolerances if no explicit values are set in the ``JointTrajectory`` message.
```

**Review by fmauch — APPROVED** [review]
I checked that all links point to the correct files and briefly went over the texts. I left a couple of comments on the way that in my opinion improve readability, but since I'm not a native speaker I wouldn't consider them required, hence approving.

**Comment by christophfroehlich** [comment]
> I checked that all links point to the correct files and briefly went over the texts. I left a couple of comments on the way that in my opinion improve readability, but since I'm not a native speaker I wouldn't consider them required, hence approving.

Thanks for the proof reading. I basically merged the infos from the documentation files with them of the yaml -> adding new information is actually out of scope -> I opened an issue for that.

@bmagyar can we merge and release that with the same sync than [3.7.0](https://github.com/ros2-gbp/generate_parameter_library-release) of generate_parameter_library? I guess we should wait for the next one?

File: joint_state_broadcaster/src/joint_state_broadcaster_parameters.yaml [hunk 2]
```diff
@@ -2,18 +2,25 @@ joint_state_broadcaster:
...
   extra_joints: {
     type: string_array,
     default_value: [],
+    description: "Names of extra joints to be added to ``joint_states`` and ``dynamic_joint_states`` with state set to 0."
   }
   interfaces: {
     type: string_array,
     default_value: [],
+    description: "Parameter to support broadcasting of only specific joints and interfaces.
+    It has to be used in combination with the ``joints`` parameter."
```
[the remark this answers is not in the document]
Comment (christophfroehlich) [reply joint_state_broadcaster/src/joint_state_broadcaster_parameters.yaml:23]: If you have a text suggestion: feel free to add one. In the meantime, I opened an issue from that ;)

**Review by bmagyar — APPROVED** [review]

**Pull request merged**